bcipy/gui/experiments/ExperimentRegistry.py

```diff
@@ -31,14 +31,118 @@ def __init__(self, *args, **kwargs):
         self.name_input = None
         self.summary_input = None
         self.field_input = None
+        self.panel = None
+        self.line_items = None
 
         self.fields = []
         self.name = None
         self.summary = None
 
+        # for registered fields
+        self.build_scroll_area()
+
         self.show_gui()
         self.update_field_list()
 
+    def build_scroll_area(self) -> None:
+        """Build Scroll Area.
+
+        Appends a scrollable area at the bottom of the UI for management of registered fields via LineItems.
+        """
+        line_widget = LineItems([], self.width)
+        self.panel = ScrollableFrame(200, self.width, background_color='white', widget=line_widget)
+        self.add_widget(self.panel)
+
+    def refresh_field_panel(self) -> None:
+        """Refresh Field Panel.
+
+        Reconstruct the line items from the registered fields and refresh the scrollable panel of registered fields.
+        """
+        self.build_line_items_from_fields()
+        self.panel.refresh(self.line_items)
+
+    def toggle_required_field(self) -> None:
+        """Toggle Required Field.
+
+        *Button Action*
+
+        Using the field_name retrieved from the button (get_id), find the field in self.experiment_fields and toggle
+            the required field ('true' or 'false').
+        """
+        field_name = self.window.sender().get_id()
+        for field in self.experiment_fields:
+            if field_name in field:
+                required = field[field_name]['required']
+                if required == 'false':
+                    field[field_name]['required'] = 'true'
+                else:
+                    field[field_name]['required'] = 'false'
```

Comment:
There are a couple places in this class where you are checking the 'required' field against a string value ('false' or 'true'). It may simplify your code if you convert this field to a boolean value when you load the experiment fields, and then convert it back to a string when you serialize the values. This code could then be simplified to:
    
    field[field_name]['required'] = not field[field_name]['required']

Reply:
I could see either way! The benefit of this approach is we don't have to loop over the experiment fields on save (it's formatted in the correct structure). This isn't trivial as it's a dict of dicts. Whereas, already have to loop over this struct during the field operations. I'll leave it here for now but I'm open to changing